src/cluster/cluster.go

```diff
@@ -1,51 +1,62 @@
 package cluster
 
 import (
-	"errors"
-	"firecontroller/component"
 	"firecontroller/microcontroller"
 	mc "firecontroller/microcontroller"
 	"firecontroller/utilities"
 	"log"
-	"strconv"
 
 	"github.com/spf13/viper"
 )
 
 //Cluster - This object defines an array of microcontrollers
 type Cluster struct {
-	Name                  string
-	SlaveMicrocontrollers []mc.Microcontroller
-	Master                *mc.Microcontroller
-	Me                    *mc.Microcontroller
+	Name             string
+	Microcontrollers []mc.Microcontroller
+	Me               *mc.Microcontroller
 }
 
 //Config -
 type Config struct {
-	Name                  string `yaml:"Name"`
-	SlaveMicrocontrollers []mc.Config
-	Master                mc.Config
+	Name             string `yaml:"Name"`
+	Microcontrollers []mc.Config
 }
 
 //GetConfig -
 func (c Cluster) GetConfig() (config Config) {
 	config.Name = c.Name
-	config.Master = c.Master.GetConfig()
-	config.SlaveMicrocontrollers = make([]mc.Config, len(c.SlaveMicrocontrollers))
-	for i, micro := range c.SlaveMicrocontrollers {
-		config.SlaveMicrocontrollers[i] = micro.GetConfig()
+	config.Microcontrollers = make([]mc.Config, len(c.Microcontrollers))
+	for i, micro := range c.Microcontrollers {
+		config.Microcontrollers[i] = micro.GetConfig()
 	}
+
 	return
 }
 
+//Master - returns a pointer to the Master micro
+func (c *Cluster) Master() *mc.Microcontroller {
+	for _, micro := range c.Microcontrollers {
+		if micro.Master {
+			return &micro
+		}
+	}
+	return &mc.Microcontroller{}
+}
+
 //Load -
 func (c *Cluster) Load(config Config) {
 	c.Name = config.Name
-	c.Master.Load(config.Master)
-	c.SlaveMicrocontrollers = make([]mc.Microcontroller, len(config.SlaveMicrocontrollers))
-	for i, micro := range config.SlaveMicrocontrollers {
-		c.SlaveMicrocontrollers[i].Load(micro)
+	newPeers := []mc.Microcontroller{
```

Comment:
this working variable is needed to preserve the initialized copy of c.Me as we load other micros in